From f648b99f4b7ce832e2377977719a510f2c456977 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Fri, 29 Jan 2016 01:48:13 +0800
Subject: [PATCH 0767/1240] fix: mmc: a3700: change SB SDIO GPIO bit back to
 SDIO mode

issue: all the SD cards cannot be detected.
fix: change the south bridge SDIO and GPIO enable bit
    back to SDIO mode.

Change-Id: Ic214547e59257147439191e7310a246c4455af7f
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27064
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm/cpu/armv8/armadalp/mvebu_gpio.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/cpu/armv8/armadalp/mvebu_gpio.c b/arch/arm/cpu/armv8/armadalp/mvebu_gpio.c
index bdd15cb..261b6ed 100644
--- a/arch/arm/cpu/armv8/armadalp/mvebu_gpio.c
+++ b/arch/arm/cpu/armv8/armadalp/mvebu_gpio.c
@@ -68,9 +68,9 @@ void mvebu_init_sdio(void)
 	reg_val |= (1 << MVEBU_GPIO_SDIO_VOLTAGE_OFF);
 	writel(reg_val, MVEBU_GPIO_SB_OUTPUT_EN_REG);
 
-	/* Set SDIO to GPIO mode */
+	/* Set test pin to SDIO mode */
 	reg_val = readl(MVEBU_GPIO_SB_SEL_REG);
-	reg_val |= (1 << MVEBU_GPIO_SB_SDIO_EN_OFF);
+	reg_val &= ~(1 << MVEBU_GPIO_SB_SDIO_EN_OFF);
 	writel(reg_val, MVEBU_GPIO_SB_SEL_REG);
 }
 
-- 
1.9.1

